hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccountState.java

```diff
@@ -389,9 +422,112 @@ public void setAlreadyUsedAutomaticAssociations(int alreadyUsedCount) {
 		autoAssociationMetadata = setAlreadyUsedAutomaticAssociationsTo(autoAssociationMetadata, alreadyUsedCount);
 	}
 
+	public Map<EntityNum, Long> getCryptoAllowances() {
+		return cryptoAllowances;
+	}
+
+	public void setCryptoAllowances(final Map<EntityNum, Long> cryptoAllowances) {
```

Comment:
Since the new maps being set **must** be `NavigableMap` implementations, we can change this type signature to `NavigableMap<EntityNum, Long>` or `TreeMap<EntityNum, Long>`.

Reply:
Done. Thanks !